File: hw/unda/U3/HW16MazeGenerator/HW16MazeGenerator/src/main/java/ec/edu/espe/hw16mazegenerator/controller/MazePrinter.java
package ec.edu.espe.hw16mazegenerator.controller;

import ec.edu.espe.hw16mazegenerator.model.Direction;
import ec.edu.espe.hw16mazegenerator.model.Room;

/**
 *
 * @author Mateo Unda
 */
public class MazePrinter {

    private final Room[][] rooms;

    public MazePrinter(Room[][] rooms) {
        this.rooms = rooms;
    }

    public void print() {
        int rows = rooms.length;
        int cols = rooms[0].length;

        for (int row = 0; row < rows; row++) {
            // Parte superior de cada celda (techo)
            for (int col = 0; col < cols; col++) {
                System.out.print("+");
                System.out.print(rooms[row][col].getWall(Direction.NORTH).isOpen() ? "   " : "---");
            }
            System.out.println("+");

            // Parte central: paredes laterales y contenido
            for (int col = 0; col < cols; col++) {
                if (row == 0 && col == 0) {
                    System.out.print("E"); // entrada
                } else {
                    System.out.print(rooms[row][col].getWall(Direction.WEST).isOpen() ? " " : "|");
                }

                System.out.print("   ");

                if (row == rows - 1 && col == cols - 1) {
                    System.out.print("S"); // salida
                } else if (col == cols - 1) {
                    System.out.print(rooms[row][col].getWall(Direction.EAST).isOpen() ? " " : "|");
                }
            }

            System.out.println();
        }

        // Parte inferior final (Sur)
        for (int col = 0; col < cols; col++) {
            System.out.print("+---");
        }
        System.out.println("+");
    }
}

File: hw/unda/U3/HW16MazeGenerator/controller/MazePrinter.py
from model.Direction import Direction
from model.Room import Room
from typing import List

class MazePrinter:
    def __init__(self, rooms: List[List[Room]]):
        self.rooms = rooms

    def print(self):
        rows = len(self.rooms)
        cols = len(self.rooms[0])

        for row in range(rows):
            # Parte superior de cada celda (techo)
            for col in range(cols):
                print("+", end="")
                print("   " if self.rooms[row][col].get_wall(Direction.NORTH).is_opened() else "---", end="")
            print("+")

            # Parte central: paredes laterales y contenido
            for col in range(cols):
                if row == 0 and col == 0:
                    print("E", end="")  # entrada
                else:
                    print(" " if self.rooms[row][col].get_wall(Direction.WEST).is_opened() else "|", end="")

                print("   ", end="")

                if row == rows - 1 and col == cols - 1:
                    print("S", end="")  # salida
                elif col == cols - 1:
                    print(" " if self.rooms[row][col].get_wall(Direction.EAST).is_opened() else "|", end="")
            print()

        # Parte inferior final (Sur)
        for col in range(cols):
            print("+---", end="")
        print("+")
